test/proposals/mips/mip-b05/mip-b05.sol

```diff
@@ -0,0 +1,304 @@
+import {ChainIds} from "@test/utils/ChainIds.sol";
+import {Proposal} from "@test/proposals/proposalTypes/Proposal.sol";
+import {Addresses} from "@test/proposals/Addresses.sol";
+import {JumpRateModel} from "@protocol/IRModels/JumpRateModel.sol";
+import {TimelockProposal} from "@test/proposals/proposalTypes/TimelockProposal.sol";
+import {CrossChainProposal} from "@test/proposals/proposalTypes/CrossChainProposal.sol";
+import {Comptroller} from "@protocol/Comptroller.sol";
+
+contract mipb05 is Proposal, CrossChainProposal, ChainIds, Configs {
+    string public constant name = "MIP-b05";
+    uint256 public constant timestampsPerYear = 60 * 60 * 24 * 365;
+    uint256 public constant SCALE = 1e18;
+
+    struct IRParams {
+        uint256 kink;
+        uint256 baseRatePerTimestamp;
+        uint256 multiplierPerTimestamp;
+        uint256 jumpMultiplierPerTimestamp;
+    }
+
+    constructor() {
+        _setNonce(5);
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./test/proposals/mips/mip-b05/MIP-B05.md")
+        );
+        _setProposalDescription(proposalDescription);
+    }
+
+    function _validateJRM(address jrmAddress, address tokenAddress, IRParams memory params)
+        internal
+    {
+        JumpRateModel jrm = JumpRateModel(jrmAddress);
+        assertEq(
+            address(MToken(tokenAddress).interestRateModel()),
+            address(jrm),
+            "interest rate model not set correctly"
+        );
+
+        assertEq(jrm.kink(), params.kink, "kink verification failed");
+        assertEq(
+            jrm.timestampsPerYear(),
+            timestampsPerYear,
+            "timestamps per year verifiacation failed"
+        );
+        assertEq(
+            jrm.baseRatePerTimestamp(),
+            (params.baseRatePerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "base rate per timestamp validation failed"
+        );
+        assertEq(
+            jrm.multiplierPerTimestamp(),
+            (params.multiplierPerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "multiplier per timestamp validation failed"
+        );
+        assertEq(
+            jrm.jumpMultiplierPerTimestamp(),
+            (params.jumpMultiplierPerTimestamp * SCALE) /
+                timestampsPerYear /
+                SCALE,
+            "jump multiplier per timestamp validation failed"
+        );
+    }
+
+    function _validateCF(Addresses addresses, address tokenAddress, uint256 collateralFactor) internal {
+        address unitrollerAddress = addresses.getAddress("UNITROLLER");
+        Comptroller unitroller = Comptroller(unitrollerAddress);
+
+        (bool _listed, uint256 collateralFactorMantissa) = unitroller.markets(tokenAddress);
+
+        assertEq(
+            collateralFactorMantissa,
+            collateralFactor,
+            "collateral factor validation failed"
+        );
+    }
+
+    function deploy(Addresses addresses, address) public override {}
```

Comment:
usually the deploy step will have deployment of the contracts for quick validation that things are deployed correctly.